src/utils/helpers.ts

```diff
@@ -235,3 +237,57 @@ export function isValidFunctionInfo(functionInfo: string): boolean {
 export function truncateAddress(address: string, start: number = 6, end: number = 5) {
   return `${address.slice(0, start)}...${address.slice(-end)}`;
 }
+
+/**
+ * Constants for metadata address calculation
+ */
+const APTOS_COIN_TYPE_STR = "0x1::aptos_coin::AptosCoin";
+const APT_METADATA_ADDRESS_HEX = AccountAddress.A.toStringLong();
+
+/**
+ * Helper function to standardize Move type string by converting all addresses to short form,
+ * including addresses within nested type parameters
```

Comment:
Why are we using short form for FAs? I thought we agreed on using the same format across products?

Reply:
This is what was hashed on-chain, so it's stuck like that.  The move to string does short form :(

Reply:
This is for offline computation of the paired metadata address. if you ask me try the algorithm is like this, it is due to the type_name implementation of David.